src/platform_impl/macos/window_delegate.rs

```diff
@@ -409,6 +409,17 @@ declare_class!(
                 self.queue_event(WindowEvent::ThemeChanged(theme));
             }
         }
+
+        #[sel(windowDidChangeScreen:)]
+        fn window_did_change_screen(&self, _: Option<&Object>) {
+            trace_scope!("windowDidChangeScreen:");
+            let shared_state = self.window.lock_shared_state("window_did_change_screen");
+            if shared_state.is_simple_fullscreen {
```

Comment:
That way you'll drop the lock right away.

```suggestion
            let is_simple_fullscreen = self.window.lock_shared_state("window_did_change_screen").is_simple_fullscreen;
            if is_simple_fullscreen {
```